back/src/domains/establishment/adapters/InMemoryDiscussionRepository.ts

```diff
@@ -13,19 +13,33 @@ export class InMemoryDiscussionRepository implements DiscussionRepository {
 
   public discussionCallsCount = 0;
 
-  public async getDiscussions({
-    createdSince,
-    sirets,
-  }: GetDiscussionsParams): Promise<DiscussionDto[]> {
+  public async getDiscussions(
+    { createdSince, sirets, lastAnsweredByCandidate }: GetDiscussionsParams,
+    limit: number,
+  ): Promise<DiscussionDto[]> {
     this.discussionCallsCount++;
     const filters: Array<(discussion: DiscussionDto) => boolean> = [
-      ({ siret }) => (sirets ? sirets.includes(siret) : true),
+      ({ siret }) =>
+        sirets && siret.length > 0 ? sirets.includes(siret) : true,
       ({ createdAt }) =>
         createdSince ? new Date(createdAt) >= createdSince : true,
+      ({ exchanges }) => {
+        if (!lastAnsweredByCandidate) return true;
+        const mostRecentExchange = exchanges.reduce((a, b) =>
+          a.sentAt > b.sentAt ? a : b,
+        );
+        const sendAt = new Date(mostRecentExchange.sentAt);
+
+        return (
+          mostRecentExchange.sender === "potentialBeneficiary" &&
+          sendAt >= lastAnsweredByCandidate.from &&
+          sendAt <= lastAnsweredByCandidate.to
+        );
+      },
     ];
-    const discussions = this.discussions.filter((discussion) =>
-      filters.every((filter) => filter(discussion)),
-    );
+    const discussions = this.discussions
+      .filter((discussion) => filters.every((filter) => filter(discussion)))
+      .filter((_, index) => index < limit);
```

Comment:
ca me paraitrait plus claire avec un slice:

```suggestion
      .slice(0, limit)
```